frontend/src/metabase/core/components/ColorRangeSelector/ColorRangePopover.tsx

```diff
@@ -44,30 +44,37 @@ const ColorSelectorContent = forwardRef(function ColorRangeSelector(
     return customColorMapping ?? getDefaultColorMapping(colors);
   }, [colors, customColorMapping]);
 
+  const [isInverted, setIsInverted] = useState(() =>
+    getDefaultIsInverted(initialValue, colorMapping),
+  );
+
   const [color, setColor] = useState(() =>
     getDefaultColor(initialValue, colors, colorMapping),
   );
 
   const [value, setValue] = useState(() =>
-    getColorRange(color, colorMapping, getInverted(initialValue, colorMapping)),
+    getColorRange(color, colorMapping, isInverted),
   );
 
-  const handleSelect = useCallback(
+  const handleColorSelect = useCallback(
     (newColor: string) => {
-      setColor(newColor);
-      setValue(getColorRange(newColor, colorMapping));
-    },
-    [colorMapping],
-  );
+      const newValue = getColorRange(newColor, colorMapping, isInverted);
 
-  const handleChange = useCallback(
-    (newValue: string[]) => {
+      setColor(newColor);
+      setValue(newValue);
       onChange?.(newValue);
-      onClose?.();
```

Comment:
For a better UX we don't close the popover, it's better to leave it open so the user can seamlessly try different colors